Wikipedia/Code/EditLinkViewController.swift

```diff
@@ -0,0 +1,203 @@
+    }
+
+    @objc private func finishEditing(_ sender: UIBarButtonItem) {
+        let displayText = displayTextView.text
+        guard let linkTarget = articleURL.wmf_title else {
+            assertionFailure("Failed to extract article title from url: \(articleURL)")
+            delegate?.editLinkViewController(self, didFailToExtractArticleTitleFromArticleURL: articleURL)
+            return
+        }
+        delegate?.editLinkViewController(self, didFinishEditingLink: displayText, linkTarget: linkTarget)
+    }
+
+    @IBAction private func removeLink(_ sender: UIButton) {
+        delegate?.editLinkViewControllerDidRemoveLink(self)
+    }
+
+    @IBAction private func searchArticles(_ sender: UITapGestureRecognizer) {
+        let searchViewController = SearchViewController()
+        searchViewController.shouldSetTitleViewWhenRecentSearchesAreDisabled = false
+        searchViewController.siteURL = siteURL
+        searchViewController.shouldSetSearchVisible = false
+        searchViewController.shouldBecomeFirstResponder = true
+        searchViewController.displayType = .backVisible
+        searchViewController.areRecentSearchesEnabled = false
+        searchViewController.dataStore = SessionSingleton.sharedInstance()?.dataStore
+        searchViewController.shouldShowCancelButton = false
+        searchViewController.delegate = self
+        searchViewController.delegatesSelection = true
+        searchViewController.showLanguageBar = false
+        searchViewController.navigationItem.title = title
+        searchViewController.searchTerm = articleURL.wmf_title
+        searchViewController.search()
+        searchViewController.apply(theme: theme)
+        navigationController?.pushViewController(searchViewController, animated: true)
+    }
+
+    override func apply(theme: Theme) {
+        super.apply(theme: theme)
+        self.theme = theme
+        guard viewIfLoaded != nil else {
+            return
+        }
+        contentView.backgroundColor = theme.colors.paperBackground
+        view.backgroundColor = theme.colors.baseBackground
+        separatorViews.forEach { $0.backgroundColor = theme.colors.border }
+        displayTextLabel.textColor = theme.colors.secondaryText
+        linkTargetLabel.textColor = theme.colors.secondaryText
+        removeLinkButton.tintColor = theme.colors.destructive
+        removeLinkButton.backgroundColor = theme.colors.paperBackground
+        closeButton.tintColor = theme.colors.primaryText
+        doneButton.tintColor = theme.colors.link
+        displayTextView.textColor = theme.colors.primaryText
+        activityIndicatorView.style = theme.isDark ? .white : .gray
+    }
+}
+
+extension EditLinkViewController: ArticleCollectionViewControllerDelegate {
+    func articleCollectionViewController(_ articleCollectionViewController: ArticleCollectionViewController, didSelectArticleWith articleURL: URL, at indexPath: IndexPath) {
+        self.articleURL = articleURL
```

Comment:
Feels like this should pop the search VC as well - it was a bit confusing to have to select "Back" to confirm the change